cypress/integration/devices_page_test.js

```diff
@@ -69,5 +69,19 @@ describe('Devices page tests', () => {
       cy.get('table tbody tr').should('have.length', 1);
       cy.get('table tbody tr td:nth-child(2)').should('contain', deviceName);
     });
+
+    it.only('correctly filters by metadata', function () {
+      const metadata = this.devices.data[0].metadata
+      const key = Object.keys(metadata)[0];
+      const value = metadata[key];
+
+      cy.get('#filterMetadataKey').type(key);
+      cy.get('table tbody tr').should('have.length', 1);
+      cy.get('table tbody tr td:nth-child(2)').should('contain', key);
+
+      cy.get('#filterMetadataValue').type(value);
+      cy.get('table tbody tr').should('have.length', 1);
+      cy.get('table tbody tr td:nth-child(2)').should('contain', value);
```

Comment:
I guess you could `cy.get('#filterMetadataKey').clear()` before running this, to be sure filtering by value actually works

Reply:
this is actually intentional. I want to filter by both value AND key

Reply:
Yes, that's what I imagined. But this test would pass even if the implementation of the value filter was broken and didn't filter.
I would suggest having test cases for

1. Filtering by key
2. Filtering by value
3. Filtering by key AND value

Reply:
that's the problem with few fixture devices.
Adding more tests will not hurt the codebase